packages/opencensus-exporter-jaeger/src/jaeger.ts

```diff
@@ -0,0 +1,197 @@
+          this.addToBuffer(root, result as number);
+          for (const span of root.spans) {
+            this.addSpanToSenderBuffer(span)
+                .then(result => {
+                  this.addToBuffer(span, result as number);
+                })
+                .catch(err => {
+                  return;
+                });
+          }
+        })
+        .catch(err => {
+          return;
+        });
+
+    // Set a buffer timeout
+    this.setBufferTimeout();
+  }
+
+  /** Not used for this exporter */
+  onStartSpan(root: types.RootSpan) {}
+
+  private addToBuffer(span: types.Span, numSpans: number) {
+    if (numSpans > 0) {
+      this.successBuffer = this.successBuffer.concat(this.spanBuffer);
+      if (numSpans === this.spanBuffer.length) {
+        this.spanBuffer = [span];
+      } else {
+        this.spanBuffer = [];
+      }
+    } else {
+      this.logger.debug('adding to buffer %s', span.name);
+      this.spanBuffer.push(span);
+    }
+  }
+
+  private addSpanToSenderBuffer(span: types.Span) {
+    const thriftSpan = ThriftUtils.spanToThrift(span);
+    return new Promise((resolve, reject) => {
+      this.sender.append(thriftSpan, (numSpans: number, err?: string) => {
+        if (err) {
+          this.logger.error(`failed to add span: ${err}`);
+          reject(-1);
+        } else {
+          this.logger.debug('successful append for : %s', numSpans);
+          resolve(numSpans);
+        }
+      });
+    });
+  }
+
+  /**
+   * Publishes a list of root spans to Jaeger.
+   * @param rootSpans
+   */
+  publish(rootSpans: types.RootSpan[]) {
+    this.logger.debug('JeagerExport publishing');
+    for (const root of rootSpans) {
+      if (this.spanBuffer.indexOf(root) <= 0) {
+        this.onEndSpan(root);
+      }
+    }
+    return this.flush();
+  }
+
+  private flush() {
+    return new Promise((resolve, reject) => {
+      try {
+        this.sender.flush((numSpans: number, err?: string) => {
+          if (err) {
+            this.logger.error(`failed to flush span: ${err}`);
+            resolve(-1);
+          } else {
+            this.logger.debug('successful flush for : %s', numSpans);
+            this.successBuffer = this.successBuffer.concat(this.spanBuffer);
+            this.spanBuffer = [];
+            resolve(numSpans);
+          }
+        });
+      } catch (err) {
+        this.logger.error(`failed to flush span: ${err}`);
+        resolve(-1);
+      }
+      if (this.checkCallback) {
+        this.checkCallback(this);
+      }
+    });
+  }
+
+  close() {
+    this.sender.close();
+  }
+
+  /** Start the buffer timeout, when finished calls flush method */
+  private setBufferTimeout() {
+    this.logger.debug('JeagerExporter: set timeout');
+    if (this.timeoutSet) {
+      return;
+    }
+    this.timeoutSet = true;
+
+    setTimeout(() => {
+      if (this.spanBuffer.length === 0) {
+        return;
+      }
+      this.timeoutSet = false;
+      this.publish([]);
```

Comment:
Does it work to call `this.flush()` instead?

Reply:
Agree.